### PR TITLE
Spotaut 17229 ocean aks migration

### PR DESCRIPTION
**Pull request by pripatra**
_This is the default pull request template. You can customize it by adding a `pull_request_template.md` at the root of your repo or inside the `.github` folder._

# Jira Ticket
https://spotinst.atlassian.net/browse/SPOTAUT-17729
_Include a link to your Jira Ticket_
_Example: [JIRAISS-1234](https://spotinst.atlassian.net/browse/JIRAISS-1234)_

# Demo

_Please add a recording of the feature/bug fix in work. if you added new routes, the recording should show the request and response for each new/changed route_

# Checklist:
- [ ] I have filled relevant self assessment ([NodeJS](https://docs.google.com/forms/d/e/1FAIpQLSfl14u9AOBAmxVJ272tvO7XNuXE-EMvWaGcaRZalu1UAKB7RA/viewform), [Frontend](https://docs.google.com/forms/d/e/1FAIpQLSdiBPNKH81w_EkavihVL8Uwb0j7tP8PwJmLFYm2nCOQxz-1qw/viewform), [Backend](https://docs.google.com/forms/d/e/1FAIpQLSed_PsTJ5-XIWkFL6BSDE2AQRBVPmwc3PAmHZkUn-erzVI37Q/viewform?usp=sf_link))
- [ ] I have run ESlint on my changes and fixed all warnings and errors (**NodeJS & Frontend Services**)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have validated all the requirements in the Jira task were answered
- [ ] I have all neccessary approvals for the design/mini design of this task
- [ ] I have approved the API changes and granular permission patterns (documentation subtask) (**For public services only**)

### CODE REVIEW

File: spotinst_sdk2/models/ocean/azure/__init__.py [hunk 1]
```diff
@@ -726,3 +726,55 @@ def __init__(self, roll: Roll = none):
...
+        self.node_pool_names = node_pool_names
+        self.batch_size_percentage = batch_size_percentage
+        self.batch_min_healthy_percentage = batch_min_healthy_percentage
+        self.comment = comment
+        self.respect_pdb = respect_pdb
+        self.respect_restrict_scale_down = respect_restrict_scale_down
+        self.should_evict_standalone_pods = should_evict_standalone_pods
+        self.should_terminate_nodes = should_terminate_nodes
+
+class MigrationRequest:
```
Comment (anuragsharma-123) [inline spotinst_sdk2/models/ocean/azure/__init__.py:770]: node_names, node_pool_names fields missing.

File: spotinst_sdk2/clients/ocean/__init__.py [hunk 2]
```diff
@@ -1575,4 +1575,120 @@ def stop_roll(self, ocean_id: str, roll_id: str):
...
+
+        return formatted_response["response"]
+
+    def stop_migration(self, ocean_id: str, migration_id: str):
+        """
+        Stop an ongoing Workload Migration.
+
+        # Arguments
+        ocean_id (String): ID of the Ocean Cluster
+        migration_id (bool): The migration identifier of a specific migration
```
Comment (anuragsharma-123) [inline spotinst_sdk2/clients/ocean/__init__.py]: This is string parameter

File: spotinst_sdk2/clients/ocean/__init__.py [hunk 3]
```diff
@@ -1575,4 +1575,120 @@ def stop_roll(self, ocean_id: str, roll_id: str):
...
+            url=self.__base_ocean_cluster_url+"/"+ocean_id+"/migration/"+migration_id,
+            entity_name="ocean_azure_migration"
+        )
+
+        formatted_response = self.convert_json(
+            response, self.camel_to_underscore)
+
+        return formatted_response["response"]
+
+    def list_migrations(self, ocean_id: str = None):
```
Comment (anuragsharma-123) [inline spotinst_sdk2/clients/ocean/__init__.py]: why do we have "= None" here. This doesn't seem to be optional field.

File: spotinst_sdk2/models/ocean/azure/__init__.py (hunk 1 above)
Comment (pripatra) [reply spotinst_sdk2/models/ocean/azure/__init__.py:770]: added them

File: spotinst_sdk2/clients/ocean/__init__.py (hunk 3 above)
Comment (pripatra) [reply spotinst_sdk2/clients/ocean/__init__.py]: removed None

File: spotinst_sdk2/clients/ocean/__init__.py (hunk 2 above)
Comment (pripatra) [reply spotinst_sdk2/clients/ocean/__init__.py]: changed to String

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**